bundles/org.openhab.core.addon/src/main/java/org/openhab/core/addon/internal/xml/AddonInfoAddonsXmlProvider.java

```diff
@@ -79,8 +79,18 @@ public Set<AddonInfo> getAddonInfos(@Nullable Locale locale) {
     }
 
     private void initialize() {
+        File fileFolder = new File(folder);
+        try {
+            if (!fileFolder.isDirectory()) {
+                logger.warn("Folder '{}' does not exist", folder);
+                return;
+            }
+        } catch (SecurityException e) {
+            logger.warn("Folder '{}' security exception", folder);
```

Comment:
In what situations can that happen? I would also print out the message of the exception here.

Reply:
I guess that SecurityException would occur if there was a file access rights violation. But I could not simulate it in tests (in my case the folder is either present or absent). However I will add the stack trace to the log.